healthtools/scrapers/health_facilities.py

```diff
@@ -28,17 +18,23 @@ def __init__(self):
         self.data_archive_key = "archive/health_facilities-{}.json"
 
         self.access_token = None
+        self.total_pages = 0
 
-    def scrape_site(self):
-        self.get_token()
-        self.get_data()
+    def scrape_site(self, page_no=1):
+        if page_no == 1:
+            self.get_token()
+
+        self.get_data(page_no)
 
         if self.results:
-            self.archive_data(json.dumps(self.results))
-            self.elasticsearch_delete_docs()
+            if page_no == 1:
+                self.elasticsearch_delete_docs()
             self.elasticsearch_index(self.results_es)
```

Comment:
Do we not run the risk of loosing the data if we can't `scrape_site` on the next page?

Reply:
We actually do, because what I currently have indexes per page and archives the data in bulk when the entire scraping is complete.
I think I can raise an exception if an error occur while scraping on the next page and archive whatever data we have on ground.